### PR TITLE
update gardener template to match the new schema

### PR DESCRIPTION
**Pull request by obadrawi**

### CODE REVIEW

File: types/provider.go [hunk 1]
```diff
@@ -14,14 +14,22 @@ type Provider struct {
...
 
 const (
-	// GCP stands for the Google Cloud Platform.
-	GCP ProviderType = "gcp"
-	// Azure stands for the Microsoft Azure Cloud Computing Platform.
-	Azure ProviderType = "azure"
-	// AWS stands for Amazon Web Services.
-	AWS ProviderType = "aws"
+	// GCPProvider stands for the Google Cloud Platform provider.
+	GCPProvider ProviderType = "gcp"
```
Comment (clebs) [inline types/provider.go]: Is there a reason to change all these constants? We already defined the API like this for a reason, unless there is a need to change this, please revert it.

File: internal/operator/terraform.go [hunk 2]
```diff
@@ -185,16 +200,16 @@ func (t *Terraform) newPlatform(providerType types.ProviderType, configuration m
 	var providerName string
 
 	switch providerType {
-	case types.GCP:
+	case types.GCPProvider:
```
Comment (clebs) [inline internal/operator/terraform.go]: Please completely revert this change.

File: types/provider.go (hunk 1 above)
Comment (obadrawi) [reply types/provider.go]: done..reverted

File: internal/operator/terraform.go (hunk 2 above)
Comment (obadrawi) [reply internal/operator/terraform.go]: done

File: examples/gardener/main.go [hunk 3]
```diff
@@ -13,10 +13,10 @@ import (
 )
 
 func main() {
-	projectName := flag.String("p", "", "Gardener project name")
+	projectName := flag.String("p", "garden-omar", "Gardener project name")
```
Comment (clebs) [inline examples/gardener/main.go]: Please remove any personal or private data from the code.

File: examples/gardener/main.go [hunk 4]
```diff
@@ -13,10 +13,10 @@ import (
...
 	machineType := flag.String("m", "n1-standard-4", "GCP machine type")
-	credentials := flag.String("c", "", "Path to the credentials file")
-	secret := flag.String("s", "", "Name of the secret to access the underlying provider of gardener")
+	credentials := flag.String("c", "C:\\Users\\D074188\\.kube\\config", "Path to the credentials file")
```
Comment (clebs) [inline examples/gardener/main.go]: Please remove any personal or private data from the code.

File: examples/gardener/main.go [hunk 5]
```diff
@@ -13,10 +13,10 @@ import (
...
+	secret := flag.String("s", "my-gcp", "Name of the secret to access the underlying provider of gardener")
```
Comment (clebs) [inline examples/gardener/main.go]: Please remove any personal or private data from the code.

File: internal/gardener/gardener.go [hunk 6]
```diff
@@ -89,7 +89,7 @@ func (g *gardenerProvisioner) Credentials(cluster *types.Cluster, provider *type
 		return nil, err
 	}
 
-	s, err := k8s.CoreV1().Secrets(fmt.Sprintf("garden-%s", provider.ProjectName)).Get(fmt.Sprintf("%s.kubeconfig", cluster.Name), metav1.GetOptions{})
+	s, err := k8s.CoreV1().Secrets(provider.ProjectName).Get(fmt.Sprintf("%s.kubeconfig", cluster.Name), metav1.GetOptions{})
```
Comment (clebs) [inline internal/gardener/gardener.go]: The previous approach was already much better from a UX perspective, we can't assume that the users have to know they have to prefix their project name with "garden-"

File: examples/gardener/main.go (hunk 3 above)
Comment (obadrawi) [reply examples/gardener/main.go]: removed

File: examples/gardener/main.go (hunk 4 above)
Comment (obadrawi) [reply examples/gardener/main.go]: removed

File: examples/gardener/main.go (hunk 5 above)
Comment (obadrawi) [reply examples/gardener/main.go]: removed

File: internal/gardener/gardener.go (hunk 6 above)
Comment (obadrawi) [reply internal/gardener/gardener.go]: reverted back to old approach

File: internal/gardener/gardener.go [hunk 7]
```diff
@@ -153,6 +153,16 @@ func (g *gardenerProvisioner) validate(cluster *types.Cluster, provider *types.P
 	} else {
 		errMessage += fmt.Sprintf(errs.CannotBeEmpty, "Provider.CustomConfigurations['target_provider']")
 	}
+	if v, ok := provider.CustomConfigurations["target_profile"]; ok {
```
Comment (clebs) [inline internal/gardener/gardener.go]: Do we really need this configuration? As I can see, the possible profiles are: GCP, Azure and AWS. I guess they always match the `target_provider` (e.g. if `target_provider` is `gcp`, the `target_profile` will always be `gcpProfile`).
If this is the case, why do we need to explicitly provide the `target_profile` if we always can infer it from the `target_provider`?
Is there any case where `target_profile` and `target_provider` are not the same?

Comment (obadrawi) [reply internal/gardener/gardener.go]: removed it from required variables and inferred the value from the provider
